File: dodo-back-end/Domain/Currency.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DodoApp.Domain
{
    public class Currency
    {
        [Key]
        public int Id { get; set; }
        public int? TransactionHeaderId { get; set; }
        [Required]
        public int ProfitAmount { get; set; }
        [Required]
        public int FundAmount { get; set; }
        [Required]
        public int ChangingProfitAmount { get; set; }
        [Required]
        public int ChangingFundAmount { get; set; }
        [Required]
        public DateTime DateOfChange { get; set; }
        public string ChangeDescription { get; set; }

        [ForeignKey(nameof(TransactionHeaderId))]
        public virtual GoodsTransactionHeader TheGoodsTransactionHeader { get; set; }
    }
}

File: dodo-front-end/src/models/currency.ts
import { ITransactionHeader } from './transaction';

export interface ICurrency {
  id?: number;
  transactionHeaderId?: number;
  profitAmount?: number;
  fundAmount?: number;
  changingProfitAmount?: number;
  changingFundAmount?: number;
  dateOfChange?: Date;
  changeDescription?: string;
  theGoodsTransactionHeader?: ITransactionHeader;
}

export interface ICurrencySummaryPerDay {
  day?: Date;
  fundAmount?: number;
  profitAmount?: number;
  totalFundChange: number;
  totalProfitChange: number;
}
